src/Smearing.jl

```diff
@@ -23,6 +23,31 @@ Derivative of the occupation function, approximation to minus the delta function
 """
 occupation_derivative(S::SmearingFunction, x) = ForwardDiff.derivative(x -> occupation(S, x), x)
 
+"""
+(f(x) - f(y))/(x - y), computed stably in the case where x and y are close
+"""
+function occupation_divided_difference(S::SmearingFunction, x, y, εF, T)
+    if T == 0
+        if x == y
+            zero(x)
+        else
+            fx = x < εF ? 1 : 0
+            fy = y < εF ? 1 : 0
+            (fx-fy)/(x-y)
+        end
+    else
+        f(z) = occupation(S, (z-εF)/T)
+        fp(z) = occupation_derivative(S, (z-εF)/T)/T
```

Comment:
I went with a compromise, the idea again being to use spaces as a visual aid for what should be grouped together rather than pursue consistency. If you still don't like it I can change it

Reply:
no that's fine.